### ISSUE TITLE
Can't call createSyncDescriptor for service with more than 8 dependencies

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#118088]
I added a service injection dependency to some class and ended up with a very confusing TS error from a line that called `createSyncDescriptor`. It turns out that you can't call it for a service with more than 8 dependencies because of how `CreateSyncFunc` is defined. It's not a problem for me anymore because that function is barely used anywhere and I just changed to using the SyncDescriptor constructor directly. I wonder whether we should delete it or fix the definition of `CreateSyncFunc` to have a definition that takes arbitrary #s of arguments, but I don't know what the intent of this code is. You touched this at some point @jrieken maybe you know something about it.

https://github.com/microsoft/vscode/blob/4bd295b6e47bef7b2dc163f8dd6ef0cb33852bf6/src/vs/platform/instantiation/common/descriptors.ts#L77-L77

**Comment by jrieken** [comment]
This should be more or less adding one more of those. Tho, maybe there is newer, fancier ways in TS nowadays

https://github.com/microsoft/vscode/blob/4bd295b6e47bef7b2dc163f8dd6ef0cb33852bf6/src/vs/platform/instantiation/common/descriptors.ts#L134

**Comment by roblourens** [comment]
That helper is only used in 1 place though. And we call `new SyncDescriptor` directly in 200+ places. Just pointing that out too.

**Comment by jrieken** [comment]
> That helper is only used in 1 place though. 

Assigning to @bpasero who is that one user of [`createSyncDescriptor`](https://github.com/microsoft/vscode/blob/9510b91e7e6645d2ad72f92af00e8ee7a7178e13/src/vs/platform/actions/common/actions.ts#L485)

**Comment by bpasero** [comment]
What am I supposed to do?

**Comment by jrieken** [comment]
tldr; you are the only user of `createSyncDescriptor`, either remove your usage or take ownership, e.g keep in line with its base class

**Comment by bpasero** [comment]
I do not recall adding this method and it is outside of the history of GH, but it seems @alexdima introduced it so I let him decide to "either remove your usage or take ownership".

The easiest is probably to inline it.
